### DISCUSSION TITLE
btn-block with btn-square should be keep the 1:1 ratio

### DISCUSSION THREAD

**Discussion by deviprsd** [saadeghi/daisyui#2446] [Ideas / feature requests]
I want to be able to responsive buttons, for example inside a grid. Since grid can fluctuate with different widths, it is important to want to keep it constrained in within its ratio. I chose square but btn-square doesn't keep its ratio when used with btn-block.

**Comment by saadeghi** [comment]
So, you want the button to fill the width and grows as a square when there's more space available?

**Reply by deviprsd** [reply]
Correct, especially when square or other shapes guarantee a certain aspect ratio

**Reply by deviprsd** [reply]
Should I make a issue?

**Reply by saadeghi** [reply]
No. That's not something were planning to add in the library as a default,  
But you can do something like this using utility classes:  
```html
<button class="btn aspect-square h-auto btn-block">Button</button>
```
https://play.tailwindcss.com/DAEQq86VFb
